src/test/javascript/spec/component/apollon-diagrams/apollon-diagram-detail.component.spec.ts

```diff
@@ -124,6 +124,25 @@ describe('ApollonDiagramDetail Component', () => {
         fixture.componentInstance.ngOnDestroy();
     });
 
+    it('validateGeneration', async () => {
+        const nonInteractiveModel = { ...model, interactive: { ...model.interactive, elements: [] } };
+
+        // setup
+        const div = document.createElement('div');
+        fixture.componentInstance.editorContainer = new ElementRef(div);
+        fixture.componentInstance.apollonDiagram = diagram;
+        fixture.componentInstance.initializeApollonEditor(nonInteractiveModel);
+        const errorSpy = jest.spyOn(alertService, 'error');
+
+        // test
+        await addDelay(500);
```

Comment:
```suggestion
        await fixture.componentInstance.apollonEditor?.nextRender
```
This should be the better way to do this. Documentation of `nextRender(): Promise<void>`:
> Returns a Promise that resolves when the current React render cycle is finished.

I tested this locally and it works and is faster (and on slower machines probably also more reliable) than using `addDelay`.
You might also want to consider replacing the other instances of addDelay in this (and maybe other) files.

Reply:
Ah thank you, I should have read the Apollon Docs. But yes that's way cleaner, please have another look at the PR. 

I removed the addDelay Function and replaced it wherever it was used.

Reply:
Apollon API for reference: https://apollon-library.readthedocs.io/en/stable/user/api.html#apollon-editor-api